### PR TITLE
Observability as code 

### PR DESCRIPTION
**Pull request by NarmathaBala**

### CODE REVIEW

**Review by liliankasem — APPROVED** [review]

File: observability/readme.md [hunk 1]
```diff
@@ -27,6 +28,20 @@ Building observable systems enables development teams at CSE to measure how well
...
 Read on how Observability can be implemented in [Machine Learning](observability/ml-observability.md) engagements effectively during Model tuning, experimentation and deployment.
 
+## Observability As Code
+
+As much as possible, configuration and management of observability assets such as cloud resource provisioning, monitoring alerts and dashboards must be managed as code. Infrastructure as Code is acheived using any one of Terraform / Ansible / ARM Templates
+
+### Why
+
+* Repeatable and automatable, avoids manual configuration of monitoring alerts and dashboards from scratch across environments.
+
```
Comment (TessFerrandez) [inline observability/readme.md:38]:
```suggestion
```

File: observability/readme.md [hunk 2]
```diff
@@ -27,6 +28,20 @@ Building observable systems enables development teams at CSE to measure how well
...
+* Configured dashboards help troubleshoot errors during integration and deployment (CI/CD)
+
```
Comment (TessFerrandez) [inline observability/readme.md:40]:
```suggestion
```

File: observability/readme.md [hunk 3]
```diff
@@ -27,6 +28,20 @@ Building observable systems enables development teams at CSE to measure how well
...
+* Audit changes and makes it possible to roll-back if there are any issues.
+
```
Comment (TessFerrandez) [inline observability/readme.md:42]:
```suggestion
```

File: observability/readme.md [hunk 4]
```diff
@@ -27,6 +28,20 @@ Building observable systems enables development teams at CSE to measure how well
 
 Read on how Observability can be implemented in [Machine Learning](observability/ml-observability.md) engagements effectively during Model tuning, experimentation and deployment.
 
+## Observability As Code
+
+As much as possible, configuration and management of observability assets such as cloud resource provisioning, monitoring alerts and dashboards must be managed as code. Infrastructure as Code is acheived using any one of Terraform / Ansible / ARM Templates
+
+### Why
+
+* Repeatable and automatable, avoids manual configuration of monitoring alerts and dashboards from scratch across environments.
```
Comment (TessFerrandez) [inline observability/readme.md:37]:
```suggestion
* It makes configuration repeatable and automatable, and avoids manual configuration of monitoring alerts and dashboards from scratch across environments.
```

File: observability/readme.md [hunk 5]
```diff
@@ -27,6 +28,20 @@ Building observable systems enables development teams at CSE to measure how well
...
+
+* Configured dashboards help troubleshoot errors during integration and deployment (CI/CD)
```
Comment (TessFerrandez) [inline observability/readme.md:39]: Why is this better with Observability as code than manually configured observability?

File: observability/readme.md [hunk 6]
```diff
@@ -27,6 +28,20 @@ Building observable systems enables development teams at CSE to measure how well
...
+
+* Audit changes and makes it possible to roll-back if there are any issues.
```
Comment (TessFerrandez) [inline observability/readme.md:41]:
```suggestion
* We can audit changes and roll them back if there are any issues.
```

File: observability/readme.md [hunk 7]
```diff
@@ -27,6 +28,20 @@ Building observable systems enables development teams at CSE to measure how well
...
+
+* Identify actionable insights from the generated metrics data across all environments, not just production.
```
Comment (TessFerrandez) [inline observability/readme.md:43]: Why is this better with observability as code rather than manual configuration?

**Review by TessFerrandez — COMMENTED** [review]
Thanks for this contribution - I had some language suggestions and some questions on things I didn't fully understand.

File: observability/readme.md (hunk 1 above)
Comment (NarmathaBala) [reply observability/readme.md:38]: addressed in the check-in

File: observability/readme.md [hunk 8]
```diff
@@ -12,7 +12,8 @@ Building observable systems enables development teams at CSE to measure how well
 ## Sections
 
 - [Pillars of Observability](#pillars-of-observability)
-- [Observability in Machine Learning](ml-observability.md)
+- [Observability in Machine Learning](observability/ml-observability.md)
```
Comment (zoekdestep) [inline observability/readme.md:15]: Maybe I'm misunderstanding so correct me if I'm wrong - this link wasn't broken, but with the new change it gives me a 404.

File: observability/readme.md [hunk 9]
```diff
@@ -27,6 +28,26 @@ Building observable systems enables development teams at CSE to measure how well
 
 Read on how Observability can be implemented in [Machine Learning](ml-observability.md) engagements effectively during Model tuning, experimentation and deployment.
 
+## Observability As Code
+
+As much as possible, configuration and management of observability assets such as cloud resource provisioning, monitoring alerts and dashboards must be managed as code. Infrastructure as Code is acheived using any one of Terraform / Ansible / ARM Templates
```
Comment (zoekdestep) [inline observability/readme.md:33]:
```suggestion
As much as possible, configuration and management of observability assets such as cloud resource provisioning, monitoring alerts and dashboards must be managed as code. Infrastructure as Code is achieved using any one of Terraform / Ansible / ARM Templates
```

Comment (zoekdestep) [inline observability/readme.md:33]: Infrastructure as code or observability as code? IaC now comes a bit out of the blue for me.

**Review by zoekdestep — COMMENTED** [review]
Thank you! Some additional thoughts from my side.

**Comment by NarmathaBala** [comment]
> Thank you! Some additional thoughts from my side.

Thanks for the review. I've addressed the changes

**Comment by zoekdestep** [comment]
@NarmathaBala you closed this without merging?

**Comment by NarmathaBala** [comment]
> @NarmathaBala you closed this without merging?

Interesting, it was an auto close

**Review by zoekdestep — APPROVED** [review]

**Comment by Guilhermeslucas** [comment]
Should we add a small list of things on the observability topic that can be written as code? For example, if we add a list with following items:

- Dashboards
- Alerts
- Log Queries

Does that make sense?

**Comment by NarmathaBala** [comment]
> Should we add a small list of things on the observability topic that can be written as code? For example, if we add a list with following items:
> 
> * Dashboards
> * Alerts
> * Log Queries
> 
> Does that make sense?

hi! Apologize, I did see this right after I squashed/merged. I think this is a good idea for a separate PR.Can you create an isse with this and we can track it better that way

**Pull request merged**